proxy/http2/Http2ConnectionState.cc

```diff
@@ -229,6 +229,11 @@ rcv_headers_frame(Http2ConnectionState &cstate, const Http2Frame &frame)
     }
   }
 
+  // Ignoring HEADERS frame on a closed stream.  The HdrHeap has gone away and it will core.
+  if (stream->get_state() == Http2StreamState::HTTP2_STREAM_STATE_CLOSED) {
+    return Http2Error(Http2ErrorClass::HTTP2_ERROR_CLASS_NONE);
```

Comment:
If the state is correct, it looks like we MUST send RST_STREAM/GOAWAY with STREAM_CLOSED (0x5).

> An endpoint MUST NOT send frames other than PRIORITY on a closed
      stream.  An endpoint that receives any frame other than PRIORITY
      after receiving a RST_STREAM MUST treat that as a stream error
      (Section 5.4.2) of type STREAM_CLOSED.  Similarly, an endpoint
      that receives any frames after receiving a frame with the
      END_STREAM flag set MUST treat that as a connection error
      (Section 5.4.1) of type STREAM_CLOSED, unless the frame is
      permitted as described below.

https://tools.ietf.org/html/rfc7540#section-5.1

Reply:
I agree with Masaori. New streams can't be CLOSED state.

Reply:
The client hasn't sent a RST_STREAM frame.  My previous commit tracked to see if a client was sending a RST_STREAM frame and would send back an error STREAM_CLOSED, but that case was not being triggered.

Reply:
It looks like the buggy client sent HEADERS frame after sending END_STREAM flag. In that case, we MUST send GOAWAY with STREAM_CLOSED.